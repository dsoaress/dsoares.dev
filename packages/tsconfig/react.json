{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "lib": ["ES2015"],
    "module": "ESNext",
    "target": "es6"
  }
}
