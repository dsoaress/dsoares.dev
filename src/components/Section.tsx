import cn from 'classnames'
import type { HTMLAttributes, ReactNode } from 'react'

export function Section({ className, ...props }: HTMLAttributes<HTMLDivElement>) {
  return <section className={cn('my-20', className)} {...props} />
}
