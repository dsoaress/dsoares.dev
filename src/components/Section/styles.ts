import tw, { styled } from 'twin.macro'

export const Section = styled.section`
  ${tw`my-20`}
`
