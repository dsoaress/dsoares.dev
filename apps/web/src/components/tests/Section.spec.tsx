import { render, screen } from '@testing-library/react'

import { Section } from '../Section'

describe('<Section />', () => {
  it('should matches snapshot', () => {
    render(<Section>Children</Section>)
    const element = screen.getByText('Children')

    expect(element).toMatchSnapshot()
  })

  it('should render with children', () => {
    render(<Section>Children</Section>)
    const element = screen.getByText('Children')

    expect(element).toHaveTextContent('Children')
  })

  it('should render with with additional className', () => {
    render(<Section className="test">Children</Section>)
    const element = screen.getByText('Children')

    expect(element).toHaveClass('test')
  })
})
