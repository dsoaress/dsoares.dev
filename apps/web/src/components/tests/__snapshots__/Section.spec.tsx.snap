// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should matches snapshot 1`] = `
<section
  class="my-20"
>
  Children
</section>
`;
